tests/projects/test_project.py

```diff
@@ -443,6 +443,19 @@ def test_set_func_requirements():
     ]
 
 
+def test_set_function_mask_credentials():
+    project = mlrun.projects.MlrunProject("newproj")
+    describe_func = mlrun.import_function("hub://describe", project=project.name)
+    describe_func.set_env("V3IO_ACCESS_KEY", "1234")
```

Comment:
This is one way this use-case can happen, but I suggest you add another test that uses `func.apply()` with `mount_v3io`, `v3io_cred` and `mount_s3` - these are common function pod modifiers (that happen as part of auto-mount as well) that will stick credentials in the function spec, and we want to verify you fix handles them.